src/include/indexlr-workers.h

```diff
@@ -338,16 +332,22 @@ MinimizeWorker::work()
 
 			auto minimizers = getMinimizers(hashes, w);
 
-			if (stlfr) {
-				std::regex right("^.*#+");
-				std::regex left("/.*");
-
-				read.barcode = std::regex_replace(read.id, right, "");
-				read.barcode = std::regex_replace(read.barcode, left, "");
+			if (read.id.find("BX:Z:") == std::string::npos) {
+				size_t rightPos = read.id.find('#');
+				if (rightPos == std::string::npos) {
+					ss << read.barcode;
+				} else {
+					size_t leftPos = read.id.find('/', rightPos + 1);
```

Comment:
```suggestion
					size_t leftPos = read.id.rfind('/');
```

Reply:
Shouldn't `leftPos` be `rightPos`? Better to call it `slashPos`.

Reply:
Yeah, mixed up left and right. I was wondering what is better a name to call that variable. Thanks for the suggestion

Reply:
And, I didn't want to use `rfind` because that gives me the last occurance, and if there are optional comments with forward slash, that would not work

Reply:
I believe (but you'll want to confirm) that `read.id` doesn't contain any sequence after the first white space.

Reply:
You are correct about `read.id` and I'm now using `rfind`.